Avoid file conflict with VTK by appending a suffix to the library.
See: https://github.com/msys2/MINGW-packages/issues/22149

diff -urN ParaView-v5.13.1/VTK/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/thirdparty/diy/vtkmdiy/CMakeLists.txt.orig ParaView-v5.13.1/VTK/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/thirdparty/diy/vtkmdiy/CMakeLists.txt
--- VTK/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/thirdparty/diy/vtkmdiy/CMakeLists.txt
+++ VTK/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/thirdparty/diy/vtkmdiy/CMakeLists.txt
@@ -176,6 +176,15 @@
     if (use_mpi AND TARGET MPI::MPI_CXX)
         target_link_libraries(${lib_name} PRIVATE MPI::MPI_CXX)
     endif()
+
+    set(_lib_output_name "${lib_name}")
+    if (_vtk_build_LIBRARY_NAME_SUFFIX)
+      string(APPEND _lib_output_name "-${_vtk_build_LIBRARY_NAME_SUFFIX}")
+    endif ()
+
+    set_target_properties(${lib_name}
+      PROPERTIES
+        OUTPUT_NAME "${_lib_output_name}")
 endfunction()
 
 # create the targets
